src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs

```diff
@@ -1519,7 +1519,23 @@ internal static string GetWinrmPluginShellName()
             // PowerShell Core uses a versioned directory to hold the plugin
             // TODO: This should be PSVersionInfo.PSVersionName once we get
             // closer to release. Right now it doesn't support alpha versions.
-            return System.String.Concat("PowerShell.", PSVersionInfo.GitCommitId);
+            string version = PSVersionInfo.GitCommitId;
+
+            // We want the resulting endpoint name to be: PowerShell.6
+            // However, the GitCommitId has a `v` like `v6.1.0`, so we need to remove
+            // the `v`
+            if (version.StartsWith("v"))
+            {
+                version = version.Substring(1);
+            }
+
+            string shellName = System.String.Concat("PowerShell.", version);
+            if (version.Contains("preview"))
+            {
+                shellName += "-Preview";
+            }
+
+            return shellName;
```

Comment:
We build `GitCommitId` in `PSVersionInfo` - so maybe build the `shellName` there too? This can simplify the script below too.

Reply:
@iSazonov that's a great suggestion.  The alternative is that we add the `v` to `install-powershellremoting.ps1` for the version specific endpoint to make it predictable but inconsistent with the generic `6` and `6-preview` endpoints.  Let's discuss this in the issue.

Reply:
Adding or removing `v` to preview isn't problem but breaking change for release. I believe it is better to use the same code for both versions and the same formatting for both endpoint names.

Reply:
I believe that it is better to fix the preview version - this is acceptable for version update 6.0 -> 6.1.